### DISCUSSION TITLE
Help with TCR-seq alignment rate

### DISCUSSION THREAD

**Discussion by seoyeonoh12** [milaboratory/mixcr#1696] [General]
Hi 

I performed TCR-seq with takara bio SMART-seq with umi and ran the MiXCR pipeline.
The sequencing seems to be undersequenced ~3e+5 reads.
The alignment rate varied from 30% to 75%. **What is the reasonable range you could say "good quality" for TCR-seq?** 
A big part of unaligned reads were **"No hits (not TCR/IG)" and "No V hits".**
I also checked **BLAST** on unaligned reads, and BLAST gave results were ncRNA gene of TRBJ-TRBC, TCR beta partical cds, corona virus gene etc....
**Is this because of low sequencing depth? or some other errors/mistakes running the pipeline?**

I used the following command. 

    mixcr analyze takara-human-rna-tcr-umi-smartseq \
    ${read1}  \
    ${read2} \
    ${sample}

The following are examples of align.reports I got (one with 76.6% rate and one with 52.1%)

Analysis time: 22.18s
Total sequencing reads: 226885
Successfully aligned reads: 173823 (76.61%)
Coverage (percent of successfully aligned):
  CDR3: 172736 (99.37%)
  FR3_TO_FR4: 5868 (3.38%)
  CDR2_TO_FR4: 4764 (2.74%)
  FR2_TO_FR4: 198 (0.11%)
  CDR1_TO_FR4: 28 (0.02%)
  VDJRegion: 185 (0.11%)
Alignment failed: no hits (not TCR/IG?): 37740 (16.63%)
Alignment failed after alignment-aided overlap: 965 (0.43%)
Alignment failed: absence of V hits: 11543 (5.09%)
Alignment failed: absence of J hits: 1742 (0.77%)
Alignment failed: no target with both V and J alignments: 1072 (0.47%)
Overlapped: 46539 (20.51%)
Overlapped and aligned: 9302 (4.1%)
Overlapped and not aligned: 37237 (16.41%)
Alignment-aided overlaps, percent of overlapped and aligned: 0 (0%)
No CDR3 parts alignments, percent of successfully aligned: 34 (0.02%)
Partial aligned reads, percent of successfully aligned: 1053 (0.61%)
Chimeras: 4 (0%)
V gene chimeras: 635 (0.28%)
J gene chimeras: 35 (0.02%)
Paired-end alignment conflicts eliminated: 25 (0.01%)
Realigned with forced non-floating bound: 180346 (79.49%)
Realigned with forced non-floating right bound in left read: 12574 (5.54%)
Realigned with forced non-floating left bound in right read: 12574 (5.54%)
TRA chains: 90531 (52.08%)
TRA non-functional: 7363 (8.13%)
TRB chains: 83288 (47.92%)
TRB non-functional: 2729 (3.28%)
Trimming report:
  R1 reads trimmed left: 0 (0%)
  R1 reads trimmed right: 0 (0%)
  Average R1 nucleotides trimmed left: 0.0
  Average R1 nucleotides trimmed right: 0.0
Tag parsing report:
  Execution time: 0ns
  Total reads: 226885
  Matched reads: 226885 (100%)
  Projection +R1 +R2: 226885 (100%)
  For variant 0:
    For projection +R1 +R2:
      R1:Left position: 5
      R1:Right position: 151
      UMI:Left position: 0
      UMI:Right position: 12
      R2:Left position: 22
      Variants: 0
      Cost: 0
      R1 length: 146
      UMI length: 12
      R2 length: 129
======================================

Total sequencing reads: 320794
Successfully aligned reads: 167251 (52.14%)
Coverage (percent of successfully aligned):
  CDR3: 164956 (98.63%)
  FR3_TO_FR4: 10290 (6.15%)
  CDR2_TO_FR4: 6947 (4.15%)
  FR2_TO_FR4: 415 (0.25%)
  CDR1_TO_FR4: 78 (0.05%)
  VDJRegion: 820 (0.49%)
Alignment failed: no hits (not TCR/IG?): 135063 (42.1%)
Alignment failed after alignment-aided overlap: 1536 (0.48%)
Alignment failed: absence of V hits: 13098 (4.08%)
Alignment failed: absence of J hits: 3236 (1.01%)
Alignment failed: no target with both V and J alignments: 610 (0.19%)
Overlapped: 163845 (51.07%)
Overlapped and aligned: 28250 (8.81%)
Overlapped and not aligned: 135595 (42.27%)
Alignment-aided overlaps, percent of overlapped and aligned: 0 (0%)
No CDR3 parts alignments, percent of successfully aligned: 40 (0.02%)
Partial aligned reads, percent of successfully aligned: 2255 (1.35%)
Chimeras: 6 (0%)
V gene chimeras: 488 (0.15%)
J gene chimeras: 14 (0%)
Paired-end alignment conflicts eliminated: 28 (0.01%)
Realigned with forced non-floating bound: 156949 (48.93%)
Realigned with forced non-floating right bound in left read: 10804 (3.37%)
Realigned with forced non-floating left bound in right read: 10804 (3.37%)
TRA chains: 97413 (58.24%)
TRA non-functional: 10905 (11.19%)
TRB chains: 69832 (41.75%)
TRB non-functional: 2820 (4.04%)
Trimming report:
  R1 reads trimmed left: 0 (0%)
  R1 reads trimmed right: 0 (0%)
  Average R1 nucleotides trimmed left: 0.0
  Average R1 nucleotides trimmed right: 0.0
Tag parsing report:
  Execution time: 0ns
  Total reads: 320794
  Matched reads: 320794 (100%)
  Projection +R1 +R2: 320794 (100%)
  For variant 0:
    For projection +R1 +R2:
      R1:Left position: 5
      R1:Right position: 151
      UMI:Left position: 0
      UMI:Right position: 12
      R2:Left position: 22
      Variants: 0
      Cost: 0
      R1 length: 146
      UMI length: 12
      R2 length: 129
======================================

Thank you in advance!

**Comment by mizraelson** [comment]
Hi,
The main issue appears to be related to the `Alignment failed: no hits (not TCR/IG?)`. Typically, we would expect more than 80% of the reads to align successfully. Increasing the depth of sequencing will hardly not resolve the issue, as the relative proportions of aligned to non-aligned reads would likely remain consistent. You are correct to investigate the non-aligned reads. Possible causes might include contamination, primer misannealing etc. It's difficult to determine the exact issue without reviewing the actual alignments.
